### PR TITLE
Add .+= to differences from Python

### PR DESCRIPTION
**Pull request by bkamins**
Add explanation of in-place operation `.+=` as opposed to `+=`.

### CODE REVIEW

File: doc/src/manual/noteworthy-differences.md [hunk 1]
```diff
@@ -205,8 +205,8 @@ For users coming to Julia from R, these are some noteworthy differences:
     be reversed in Julia relative to NumPy (see relevant section of [Performance Tips](@ref man-performance-tips)).
   * Julia's updating operators (e.g. `+=`, `-=`, ...) are *not in-place* whereas NumPy's are. This
     means `A = ones(4); B = A; B += 3` doesn't change values in `A`, it rather rebinds the name `B`
-    to the result of the right- hand side `B = B + 3`, which is a new array. Use `B[:] += 3`, explicit
-    loops, or `InplaceOps.jl`.
+    to the result of the right- hand side `B = B + 3`, which is a new array. Use `B .+= 3`
+    (see also [dot operators](@ref man-dot-operators)), `B[:] += 3`, explicit loops, or `InplaceOps.jl`.
```
Comment (stevengj) [inline doc/src/manual/noteworthy-differences.md]: `B[:] += 3` still allocates an array unless you do `@views B[:] += 3`.

File: doc/src/manual/noteworthy-differences.md [hunk 2]
```diff
@@ -205,8 +205,8 @@ For users coming to Julia from R, these are some noteworthy differences:
     be reversed in Julia relative to NumPy (see relevant section of [Performance Tips](@ref man-performance-tips)).
   * Julia's updating operators (e.g. `+=`, `-=`, ...) are *not in-place* whereas NumPy's are. This
     means `A = ones(4); B = A; B += 3` doesn't change values in `A`, it rather rebinds the name `B`
-    to the result of the right- hand side `B = B + 3`, which is a new array. Use `B[:] += 3`, explicit
-    loops, or `InplaceOps.jl`.
+    to the result of the right- hand side `B = B + 3`, which is a new array. Use `B .+= 3`
```
Comment (Sacha0) [inline doc/src/manual/noteworthy-differences.md]: While modifying these lines, should the space exist in "right- hand"? Also might be nice to extend "Use `B .+= 3`..." to "For in-place operation, use `B .+= 3`..."?

**Review by Sacha0 — APPROVED** [review]
Thanks @bkamins! :)

File: doc/src/manual/noteworthy-differences.md (hunk 2 above)
Comment (stevengj) [inline doc/src/manual/noteworthy-differences.md]: no space... should be "right-hand side"

**Comment by Sacha0** [comment]
Absent objections, requests for time, or someone beating me to it, I plan to merge this tomorrow morning. Best!

**Comment by Sacha0** [comment]
Thanks @bkamins! :)

**Pull request merged**